papa/src/main.rs

```diff
@@ -0,0 +1,465 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![no_std]
+#![no_main]
+#![feature(asm, const_fn, lang_items, compiler_builtins_lib)]
+#![feature(in_band_lifetimes)]
+#![feature(core_intrinsics)]
+
+extern crate capsules;
+extern crate h1b;
+#[macro_use(static_init, debug, create_capability)]
+extern crate kernel;
+extern crate cortexm3;
+
+#[macro_use]
+pub mod io;
+
+pub mod digest;
+pub mod aes;
+pub mod dcrypto;
+pub mod dcrypto_test;
+pub mod debug_syscall;
+mod flash_test;
+mod nvcounter_syscall;
+mod nvcounter_test;
+pub mod personality;
+pub mod virtual_flash;
+
+use capsules::alarm::AlarmDriver;
+use capsules::console;
+use capsules::virtual_alarm::VirtualMuxAlarm;
+use capsules::virtual_uart::{MuxUart, UartDevice};
+
+
+use kernel::{Chip, Platform};
+use kernel::capabilities;
+use kernel::hil;
+use kernel::hil::entropy::Entropy32;
+use kernel::hil::rng::Rng;
+use kernel::mpu::MPU;
+
+use h1b::crypto::dcrypto::Dcrypto;
+use h1b::hil::flash::Flash;
+use h1b::nvcounter::{FlashCounter,NvCounter};
+use h1b::timels::Timels;
+use h1b::usb::{Descriptor, StringDescriptor};
```

Comment:
Do you want USB to be part of Papa? It doesn't seem relevant.

Reply:
Oops, good catch. I meant to remove that.